src/pdf_parser/pdf_utils/disambiguate_layout.py

```diff
@@ -224,25 +224,29 @@ def reduce_all_overlapping_boxes(
     Returns:
         The new layout with blocks having no overlapping coordinates.
     """
-    assert reduction_direction in ("vertical", "horizontal"), (
-        "Invalid direction. Must be 'vertical' or " "'horizontal'. "
-    )
-    for box_1 in blocks:
-        for box_2 in blocks:
-            if box_1 == box_2:
+    assert reduction_direction in (
+        "vertical",
+        "horizontal",
+    ), "Invalid direction. Must be 'vertical' or 'horizontal'."
+    edited_blocks = []
+    edited_coords = []
+    for i, box_1 in enumerate(blocks):
```

Comment:
We can make this into a function: 

      def identify_overlapping_boxes():
           ....